pkgs/build-support/setup-hooks/reproducible-builds.sh

```diff
@@ -0,0 +1,2 @@
+#export NIX_SET_BUILD_ID=${SOURCE_DATE_EPOCH:1}
+export NIX_CFLAGS_COMPILE+=" -frandom-seed=${SOURCE_DATE_EPOCH:1} "
```

Comment:
how about 
```suggestion
export NIX_CFLAGS_COMPILE+=" -frandom-seed=$out "
```

Reply:
It's a good question. We also had that idea.

Andi's argument was that it would prevent content-addressable outputs since the output will ten always change if $out changes.

Reply:
I even think that $SOURCE_DATE_EPOCH has that problem. I've not yet come up with a good idea here. Apparently, based on random googling, a single, static value that is the same across all builds isn't great. This went to the extend of being different per input file (e.g. it's hash). I am not convinced or an expert here :/

Reply:
I ended up going with the bash equivalent of `basename $out` as I want to avoid embedding a store path into the binary (and in e.g. multi-output derivations carrying a reference).